mlir/test/Dialect/Vector/canonicalize.mlir

```diff
@@ -2932,6 +2932,17 @@ func.func @contiguous_extract_strided_slices_to_extract(%arg0 : vector<8x1x2x1x1
 
 // -----
 
+// CHECK-LABEL: @contiguous_extract_strided_slices_to_extract_no_unit_dims
+// CHECK:        %[[EXTRACT:.+]] = vector.extract {{.*}}[0, 0] : vector<4xi32> from vector<8x2x4xi32>
+// CHECK-NEXT:   return %[[EXTRACT]] :  vector<4xi32>
+func.func @contiguous_extract_strided_slices_to_extract_no_unit_dims(%arg0 : vector<8x2x4xi32>) -> vector<4xi32> {
+  %1 = vector.extract_strided_slice %arg0 {offsets = [0, 0], sizes = [1, 1], strides = [1, 1]} : vector<8x2x4xi32> to vector<1x1x4xi32>
+  %2 = vector.shape_cast %1 : vector<1x1x4xi32> to vector<4xi32>
+  return %2 : vector<4xi32>
+}
```

Comment:
[nit]  I don't think that lack of "unit dims" is that significant here.

What's more important is the fact that there's no overlap between "source" leading dimensions and "sizes" (which are required to be "1" for this particular pattern).  That leads to `numOffsets` to remain 2 when on L3971 in VectorOps.cpp. And then the out-of-bounds access happens.

So, I'd be tempted to:
* `@contiguous_extract_strided_slices_to_extract` -> `@contiguous_extract_strided_slices_to_extract_sizes_and_outer_source_dims_overlap`
* `@contiguous_extract_strided_slices_to_extract_no_unit_dims ` -> `@contiguous_extract_strided_slices_to_extract_sizes_and_outer_source_dims_no_overlap`

Alternatively, add a note that `no_unit_dims` refers to "no unit dims in the source vector`. That was not clear to me.